be/src/Unic.UrlMapper2/code/Services/RedirectSearcher.cs

```diff
@@ -63,15 +63,30 @@ protected virtual Redirect MapToSearchResult(RedirectSearchResultItem redirectSe
                 this.logger.Error($"Failed to parse source protocol {redirectSearchResultItem.SourceProtocol}", this);
             }
 
-            return new Redirect
+            var redirect = new Redirect
             {
                 RedirectSearchData = redirectSearchData,
                 ItemId = redirectSearchResultItem.ItemId,
                 RedirectType = redirectType,
                 SourceProtocol = sourceProtocol,
                 RegexEnabled = redirectSearchResultItem.RegexEnabled,
+                PreserveQueryString = redirectSearchResultItem.PreserveQueryString,
                 Term = redirectSearchResultItem.SourceTerm
             };
+
+            this.HandlePreserveQueryString(redirect, redirectSearchResultItem);
+            return redirect;
+        }
+
+        protected virtual void HandlePreserveQueryString(Redirect redirect, RedirectSearchResultItem redirectSearchResultItem)
+        {
+            if (!redirect.PreserveQueryString) return;
+
+            redirect.RegexEnabled = true;
+            if (!redirectSearchResultItem.SourceTerm.EndsWith(Constants.RegexExpressions.QueryStringExpression))
+            {
+                redirect.Term = redirectSearchResultItem.SourceTerm + Constants.RegexExpressions.QueryStringExpression;
```

Comment:
**question: (non-blocking)** If the initial  `SourceTerm` already contains a regex and we enable the regex functionality as part of this new checkbox, would this break the redirect functionality?

**thoughts:** We could consider this a misconfiguration by the authors which would break the functionality. The only other option would most likely be to detect an existing regex in the `SourceTerm` and either combine it with the query string regex or remove/replace it.

Reply:
Answering your first question, it will not break the functionality.
Those examples with enabled `Preserve Query String` still works fine:
1. In first example, we will not add query string regex because it's already existing, and the redirect works as expected
![image](https://user-images.githubusercontent.com/17814211/159219295-d3a48dbc-6757-43a2-9f11-6a15d63a63b9.png)
2. In second example we have the same case like in first one, the redirect works as expected
![image](https://user-images.githubusercontent.com/17814211/159219177-8cecc798-7110-4fa4-b5d7-72a4e0b03573.png)
3. In third case we will **add** query string regex, but it will not crush existing configuration and the redirect works as expected
![image](https://user-images.githubusercontent.com/17814211/159219156-2196705b-5d69-4638-89bf-fd48a46cf541.png)

As you can see configurations are various. IMO potentially adding query string at the end of existing regex is less dangerous than replacing existing one. Of course, my proposed solution potentially also may crush existing regex, because at the end there will be a regex patter which doesn't make seanse. But still, I would say that this is less buggy then removing/replacing.

Reply:
Thank you for the explanations and examples. This seems quite robust.